webapp/src/app.tsx

```diff
@@ -87,13 +87,19 @@ const App = React.memo((): JSX.Element => {
                                 </Route>
                                 <Route
                                     path='/workspace/:workspaceId/:boardId?/:viewId?'
-                                    render={({match}) => {
+                                    render={({match, location}) => {
                                         if (loggedIn === false) {
-                                            let redirectUrl = '/' + Utils.buildURL(`/workspace/${match.params.workspaceId}/`)
+                                            let redirectUrl = '/' + Utils.buildURL(`/workspace/${match.params.workspaceId}${match.params.boardId ? `/${match.params.boardId}` : ''}${match.params.viewId ? `/${match.params.viewId}` : ''}${location.search}`)
                                             if (redirectUrl.indexOf('//') === 0) {
                                                 redirectUrl = redirectUrl.slice(1)
                                             }
-                                            const loginUrl = `/login?r=${encodeURIComponent(redirectUrl)}`
```

Comment:
This URL should be used when not running as a plugin, so Personal Server works.

Reply:
Actually, this url is the same for both, it is the same url for mattermost-webapp, so I think it is fine.